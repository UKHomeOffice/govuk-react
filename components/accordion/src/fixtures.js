import React, { useMemo, useReducer } from 'react';
import PropTypes from 'prop-types';
import Link from '@govuk-react/link';
import ListNavigation from '@govuk-react/list-navigation';
import Paragraph from '@govuk-react/paragraph';

import Accordion from '.';

const sharedPropTypes = {
  initialState: PropTypes.shape({}).isRequired,
};

const constructFull = ({ bool, state }) =>
  Object.keys(state).reduce(
    (prev, key) => ({
      ...prev,
      [key]: bool,
    }),
    {}
  );

const reducer = (state, newState) => ({ ...state, ...newState });

const useLegacyState = state => useReducer(reducer, state);

const checkOpen = state => Object.values(state).every(Boolean);

const useAccordion = initialState => {
  const [state, setState] = useLegacyState(initialState);

  const objOpen = useMemo(() => constructFull({ bool: true, state }), [initialState, state.length]);
  const objClose = useMemo(() => constructFull({ bool: false, state }), [initialState, state.length]);

  const areAllOpen = useMemo(() => checkOpen(state), [state]);

  const toggleAll = e => {
    e.preventDefault();
    return setState(!areAllOpen ? objOpen : objClose);
  };

  const individualAccordionSetState = property => () =>
    setState({
      [property]: !state[property],
    });

  return {
    areAllOpen,
    individualAccordionSetState,
    toggleAll,
    state,
  };
};

const ExampleWithoutSummaries = ({ initialState }) => {
  const { areAllOpen, individualAccordionSetState, toggleAll, state } = useAccordion(initialState);

  return (
    <Accordion>
      <Accordion.OpenAll href="#" onClick={toggleAll}>
        {`${areAllOpen ? 'Close' : 'Open'} all`}
      </Accordion.OpenAll>
      <Accordion.Group
        heading="Writing well for the web"
        expanded={state.accordionOne}
        onClick={individualAccordionSetState('accordionOne')}
      >
        <Paragraph mb={0}>This is the content for writing well for the web.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="Writing well for specialists"
        expanded={state.accordionTwo}
        onClick={individualAccordionSetState('accordionTwo')}
      >
        <Paragraph mb={0}>This is the content for writing well for the specialists.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="Know your audience"
        expanded={state.accordionThree}
        onClick={individualAccordionSetState('accordionThree')}
      >
        <Paragraph mb={0}>This is the content for Know you audience.</Paragraph>
      </Accordion.Group>
      <Accordion.Group
        heading="How people read"
        expanded={state.accordionFour}
        onClick={individualAccordionSetState('accordionFour')}
      >
        <Paragraph mb={0}>This is the content for How people read.</Paragraph>
      </Accordion.Group>
    </Accordion>
  );
};

ExampleWithoutSummaries.propTypes = sharedPropTypes;

const ExampleWithSummaries = ({ initialState }) => {
  const { areAllOpen, individualAccordionSetState, toggleAll, state } = useAccordion(initialState);

  return (
    <Accordion>
      <Accordion.OpenAll href="#" onClick={toggleAll}>
        {`${areAllOpen ? 'Close' : 'Open'} all`}
      </Accordion.OpenAll>
      <Accordion.Group
        heading="Understanding agile project management"
        summary="Introductions, methods, core features."
        expanded={state.accordionOne}
        onClick={individualAccordionSetState('accordionOne')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Agile and government services: an introduction',
            'Agile methods: an introduction',
            'Core principles of agile',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Working with agile methods"
        summary="Workspaces, tools and techniques, user stories, planning."
        expanded={state.accordionTwo}
        onClick={individualAccordionSetState('accordionTwo')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Creating an agile working environment',
            'Agile tools and techniques',
            'Set up a team wall',
            'Writing user stories',
            'Planning in agile',
            'Deciding on priorities',
            'Developing a roadmap',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Governing agile services"
        summary="Principles, measuring progress, spending money."
        expanded={state.accordionThree}
        onClick={individualAccordionSetState('accordionThree')}
      >
        <ListNavigation listStyleType="none">
          {[
            'How the discovery phase works',
            'Measuring and reporting progress',
            'Spend controls: check if you need approval to spend money on a service',
            'Spend controls: apply for approval to spend money on a service',
            'Working across organisational boundaries',
          ].map(elem => (
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
      <Accordion.Group
        heading="Phases of an agile project"
        summary="Discovery, alpha, beta, live and retirement."
        expanded={state.accordionFour}
        onClick={individualAccordionSetState('accordionFour')}
      >
        <ListNavigation listStyleType="none">
          {[
            'Governance principles for agile service delivery',
            'How the alpha phase works',
            'How the beta phase works',
            'How the live phase works',
            'Retiring your service',
          ].map(elem => (
            /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
            <Link key={elem} href={`#myprefix-${elem}`}>
              {elem}
            </Link>
          ))}
        </ListNavigation>
      </Accordion.Group>
    </Accordion>
  );
};

ExampleWithSummaries.propTypes = sharedPropTypes;

class AccordionClassComponent extends React.Component {
  constructor() {
    super();
    this.state = {
      accordionOne: false,
      accordionTwo: false,
      accordionThree: false,
      accordionFour: false,
    };
    this.individualAccordionSetState = this.individualAccordionSetState.bind(this);
    this.toggleAll = this.toggleAll.bind(this);
  }

  toggleAll() {
    return this.setState(state =>
      Object.keys(state).reduce(
        (prev, key) => ({
          ...prev,
          [key]: !this.isAllOpen,
        }),
        {}
      )
    );
  }

  individualAccordionSetState(property) {
    return () =>
      this.setState(state => ({
        [property]: !state[property],
      }));
  }

  render() {
    this.isAllOpen = checkOpen(this.state);
    return (
      <Accordion>
        <Accordion.OpenAll href="#" onClick={this.toggleAll}>
          {`${this.isAllOpen ? 'Close' : 'Open'} all`}
        </Accordion.OpenAll>
        <Accordion.Group
          heading="Writing well for the web"
          expanded={this.state.accordionOne}
          onClick={this.individualAccordionSetState('accordionOne')}
        >
          <Paragraph mb={0}>This is the content for writing well for the web.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="Writing well for specialists"
          expanded={this.state.accordionTwo}
          onClick={this.individualAccordionSetState('accordionTwo')}
        >
          <Paragraph mb={0}>This is the content for writing well for the specialists.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="Know your audience"
          expanded={this.state.accordionThree}
          onClick={this.individualAccordionSetState('accordionThree')}
        >
          <Paragraph mb={0}>This is the content for Know you audience.</Paragraph>
        </Accordion.Group>
        <Accordion.Group
          heading="How people read"
          expanded={this.state.accordionFour}
          onClick={this.individualAccordionSetState('accordionFour')}
        >
          <Paragraph mb={0}>This is the content for How people read.</Paragraph>
        </Accordion.Group>
      </Accordion>
    );
  }
}

export { AccordionClassComponent, ExampleWithoutSummaries, ExampleWithSummaries };
