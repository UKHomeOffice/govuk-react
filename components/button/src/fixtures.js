import React from 'react';
import { boolean } from '@storybook/addon-knobs/react';
import { ButtonArrow } from '@govuk-react/icons';
import Button from '.';

export default () => <Button>Save and continue</Button>;

export const ButtonStart = () => <Button start={boolean('Start', true)}>Save and continue</Button>;

export const ButtonStartIcon = () => (
  <Button icon={<ButtonArrow />} start={boolean('Start', true)}>
    Start now
  </Button>
);

export const ButtonDisabled = () => <Button disabled={boolean('Disabled', true)}>Disabled primary button</Button>;

export const ButtonDisabledStartIcon = () => (
  <Button
    disabled={boolean('Disabled', true)}
    start={boolean('Start', true)}
    icon={<ButtonArrow />}
  >
    Start now
  </Button>
);
