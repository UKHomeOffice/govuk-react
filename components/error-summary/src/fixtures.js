import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import styled from 'react-emotion';

import ErrorSummary from '.';

export const heading = 'Message to alert the user to a problem goes here';
export const description = 'Optional description of the errors and how to correct them';
export const errors = [
  'Descriptive link to the target question with an error',
  'Descriptive link to the other target question with an error',
];

const StyledDiv = styled('div')({
  marginBottom: '150px',
});

export default class ErrorSummaryExample extends React.Component {
  errors = [
    {
      id: 0,
      handleScrollToElement: () => this.targetQuestion.scrollIntoView(),
      text: 'Descriptive link to the target question with an error',
    },
    {
      id: 1,
      handleScrollToElement: () => this.otherTargetQuestion.scrollIntoView(),
      text: 'Descriptive link to the other target question with an error',
    },
  ];

  render() {
    return (
      <div>
        <ErrorSummary
          heading={text('heading', heading)}
          description={text('description', description)}
          errors={this.errors}
        />
        <StyledDiv innerRef={(node) => { this.targetQuestion = node; }}>
          Target Question
        </StyledDiv>
        <div ref={(node) => { this.otherTargetQuestion = node; }}>
          Other Target Question
        </div>
      </div>
    );
  }
}
