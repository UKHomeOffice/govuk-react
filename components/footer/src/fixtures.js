import React from 'react';
import { boolean, text } from '@storybook/addon-knobs/react';

import { BrowserRouter, Link, Route } from 'react-router-dom';

import Footer from '.';

const withRoutes = component => (
  <BrowserRouter>
    {component}
    <Route path="/footer-nav-item-2" render={() => 'Footer Navigation item 2'} />
    <Route path="/footer-meta-item-2" render={() => 'Footer Meta item 2'} />
  </BrowserRouter>
);

export default () => <Footer />;

const metaLinks = (
  <Footer.MetaLinks heading="Support links">
    <Footer.Link href="/">Item 1</Footer.Link>
    <Footer.Link to="/footer-meta-item-2" as={Link}>
      Item 2 (Router Link)
    </Footer.Link>
    <Footer.Link href="/">Item 3</Footer.Link>
  </Footer.MetaLinks>
);

const metaCustom = (
  <Footer.MetaCustom>
    Built by the <Footer.Link href="/">Government Digital Service</Footer.Link>
  </Footer.MetaCustom>
);

const metaLinksAndCustom = (
  <React.Fragment>
    {metaLinks}
    {metaCustom}
  </React.Fragment>
);

const navigation = (
  <Footer.Navigation>
    <Footer.NavigationLinks heading="Two column list" listColumns="2">
      <Footer.Link href="/">Navigation item 1</Footer.Link>
      <Footer.Link to="/footer-nav-item-2" as={Link}>
        Navigation item 2 (Router Link)
      </Footer.Link>
      <Footer.Link href="/">Navigation item 3</Footer.Link>
      <Footer.Link href="/">Navigation item 4</Footer.Link>
      <Footer.Link href="/">Navigation item 5</Footer.Link>
      <Footer.Link href="/">Navigation item 6</Footer.Link>
    </Footer.NavigationLinks>
    <Footer.NavigationLinks heading="Single column list">
      <Footer.Link href="/">Navigation item 1</Footer.Link>
      <Footer.Link href="/">Navigation item 2</Footer.Link>
      <Footer.Link href="/">Navigation item 3</Footer.Link>
    </Footer.NavigationLinks>
  </Footer.Navigation>
);

export const FooterWithMetaLinks = () => withRoutes(<Footer meta={metaLinks} />);

export const FooterWithMetaCustom = () => <Footer meta={metaCustom} />;

export const FooterWithMetaLinksAndCustom = () => withRoutes(<Footer meta={metaLinksAndCustom} />);

export const FooterWithNavigation = () => withRoutes(<Footer>{navigation}</Footer>);

export const FooterWithNavigationAndMeta = () => withRoutes(<Footer meta={metaLinksAndCustom}>{navigation}</Footer>);
