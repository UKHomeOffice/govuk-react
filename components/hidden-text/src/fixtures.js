import React from 'react';
import { text } from '@storybook/addon-knobs/react';
import Paragraph from '@govuk-react/paragraph';

import HiddenText from '.';

export const summaryText = 'Help with nationality';

export const paragraphs = [
  'If you’re not sure about your nationality, try to find out from an official document like a passport or national ID card.',
  'We need to know your nationality so we can work out which elections you’re entitled to vote in. If you can’t provide your nationality, you’ll have to send copies of identity documents through the post.',
];

export default () => (
  <HiddenText summaryText={summaryText}>
    { paragraphs.map((paragraphText, index) =>
      (index === paragraphs.length - 1
       ? <Paragraph key={`paragraphText${index + 1}`} mb={0}>{text(`paragraphText${index + 1}`, paragraphText)}</Paragraph>
       : <Paragraph key={`paragraphText${index + 1}`}>{text(`paragraphText${index + 1}`, paragraphText)}</Paragraph>
      ))}
  </HiddenText>
);
