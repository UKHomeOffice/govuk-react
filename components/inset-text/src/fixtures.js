import React from 'react';
import { boolean } from '@storybook/addon-knobs/react';
import InsetText from '.';

const insetText = 'It can take up to 8 weeks to register a lasting power of attorney if there are no mistakes in the application.';

export default () => <InsetText>{insetText}</InsetText>;

export const InsetTextNarrowBorder = () => <InsetText isNarrow={boolean('IsNarrow', true)}>{insetText}</InsetText>;
