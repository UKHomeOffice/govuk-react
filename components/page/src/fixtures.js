import React from 'react';
import { BackLink, H1 } from 'govuk-react';
import Page from '.';

const MyPage = () => (
  <Page
    beforeChildren={<BackLink href="#" />}
  >
    <H1>Page Title</H1>
  </Page>);

export default MyPage;
