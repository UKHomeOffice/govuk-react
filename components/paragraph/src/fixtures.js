import React from 'react';
import { text, boolean } from '@storybook/addon-knobs/react';
import Paragraph from '.';

const exampleText = 'A 19px body copy paragraph. [This supports markdown](https://en.wikipedia.org/wiki/Markdown) to allow for **bold** text and italics. This includes even more text to give a good representation of a more average length paragraph. That way you can see more than one line wrapping.';
const exampleSupportingText = 'A 16px supporting text paragraph. This supports markdown to allow for **bold** text and *italics*. This includes even more text to give a good representation of a more average length paragraph. That way you can see more than one line wrapping.';

export default () => <Paragraph supportingText={boolean('supportingText', false)}>{text('children', exampleText)}</Paragraph>;

export const SupportingParagraph = () => <Paragraph supportingText>{text('children', exampleSupportingText)}</Paragraph>;
