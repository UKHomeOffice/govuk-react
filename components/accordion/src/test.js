import React from 'react';
import { cleanup, fireEvent, render } from 'react-testing-library';
import { YELLOW } from 'govuk-colours';
import { FOCUS_WIDTH } from '@govuk-react/constants';

import 'jest-dom/extend-expect';

import { ExampleWithSummaries, ExampleWithoutSummaries } from './fixtures';

const focusedStyle = `
  outline: ${FOCUS_WIDTH} solid ${YELLOW};
  outline-offset: 0;
`;

afterEach(cleanup);

describe('Tabs', () => {
  it('clicking button changes focus on header', async () => {
    const { container } = render(
      <ExampleWithoutSummaries
        initialState={{
          accordionOne: false,
          accordionTwo: false,
          accordionThree: false,
          accordionFour: false,
        }}
      />
    );
    const firstButton = container.querySelector('button');
    const secondButton = container.querySelectorAll('button')[1];

    // const firstHeader = container.querySelector('div div div div:nth-child(2) div');
    const firstHeader = container.querySelector('div:nth-child(2) > div');
    const secondHeader = container.querySelector('div:nth-child(3) > div');

    expect(firstHeader).not.toHaveStyle(focusedStyle);

    // represents click event, jsdom handles focus onClick differently to browser
    fireEvent.focus(firstButton);

    expect(firstHeader).toHaveStyle(focusedStyle);

    // represents second click event, jsdom handles focus onClick differently to browser
    fireEvent.blur(firstButton);
    fireEvent.focus(secondButton);

    expect(firstHeader).not.toHaveStyle(focusedStyle);
    expect(secondHeader).toHaveStyle(focusedStyle);
  });

  it('clicking button toggles visible children', () => {
    const { container, queryByText } = render(
      <ExampleWithoutSummaries
        initialState={{
          accordionOne: true,
          accordionTwo: false,
          accordionThree: false,
          accordionFour: false,
        }}
      />
    );
    const firstButton = container.querySelector('button');

    const children = queryByText('This is the content for writing well for the web.');

    expect(children).toBeInTheDocument();

    fireEvent.click(firstButton);

    expect(children).not.toBeInTheDocument();
  });

  it('clicking close all toggles visible children', () => {
    const { container, queryByText } = render(
      <ExampleWithoutSummaries
        initialState={{
          accordionOne: true,
          accordionTwo: true,
          accordionThree: true,
          accordionFour: true,
        }}
      />
    );
    const closeAll = container.querySelector('a');

    const children = queryByText('This is the content for How people read.');

    expect(children).toBeInTheDocument();

    fireEvent.click(closeAll);

    expect(children).not.toBeInTheDocument();
  });

  it('matches wrapper snapshot', () => {
    const { asFragment } = render(
      <ExampleWithoutSummaries
        initialState={{
          accordionOne: false,
          accordionTwo: false,
          accordionThree: false,
          accordionFour: false,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('covers summary branch5', () => {
    const { asFragment } = render(
      <ExampleWithSummaries
        initialState={{
          accordionOne: false,
          accordionTwo: false,
          accordionThree: false,
          accordionFour: false,
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
