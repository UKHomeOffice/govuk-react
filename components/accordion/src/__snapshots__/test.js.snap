// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs covers summary branch5 1`] = `
<DocumentFragment>
  .c7 {
  position: absolute;
  top: 50%;
  right: 15px;
  width: 16px;
  height: 16px;
  margin-top: -8px;
}

.c7:after,
.c7:before {
  content: "";
  box-sizing: border-box;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 25%;
  height: 25%;
  margin: auto;
  border: 2px solid transparent;
  background-color: #0b0c0c;
}

.c7:before {
  width: 100%;
}

.c7:after {
  height: 100%;
}

.c2 {
  border-top: 1px solid #bfc1c3;
  padding-top: 0;
}

.c3 {
  position: relative;
  padding-right: 40px;
  padding-bottom: 15px;
  cursor: pointer;
}

.c3:hover {
  background-color: #f8f8f8;
}

.c4 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0;
  padding-top: 15px;
  padding-bottom: 0;
  padding-left: 0;
  border-width: 0;
  color: #005ea5;
  background: none;
  text-align: left;
  cursor: pointer;
}

.c5:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c5:focus {
  outline: none;
  background: none;
}

.c5:after {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c6 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  margin-top: 10px;
  margin-bottom: 0;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  display: inline;
  border-width: 0;
  color: #005ea5;
  background: none;
  cursor: pointer;
}

.c1:link {
  color: #005ea5;
}

.c1:visited {
  color: #4c2c92;
}

.c1:hover {
  color: #2b8cc4;
}

.c1:active {
  color: #2b8cc4;
}

.c1:focus {
  color: #0b0c0c;
}

.c1:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c1:focus {
  background: none;
}

.c0 {
  margin-bottom: 20px;
  border-bottom: 1px solid #bfc1c3;
}

@media (hover:none) {
  .c3:hover {
    background-color: initial;
  }
}

@media print {
  .c5 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media print {
  .c5 {
    font-family: sans-serif;
  }
}

@media print {
  .c6 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    font-family: sans-serif;
  }
}

@media print {
  .c1[href^="/"]::after,
  .c1[href^="http://"]::after,
  .c1[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

<div
    class="c0"
  >
    <div
      style="text-align: right;"
    >
      <a
        class="c1"
        href="#"
      >
        Open all
      </a>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Understanding agile project management
          </button>
        </h2>
        <div
          class="c6"
        >
          Introductions, methods, core features.
        </div>
        <div
          class="c7"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Working with agile methods
          </button>
        </h2>
        <div
          class="c6"
        >
          Workspaces, tools and techniques, user stories, planning.
        </div>
        <div
          class="c7"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Governing agile services
          </button>
        </h2>
        <div
          class="c6"
        >
          Principles, measuring progress, spending money.
        </div>
        <div
          class="c7"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Phases of an agile project
          </button>
        </h2>
        <div
          class="c6"
        >
          Discovery, alpha, beta, live and retirement.
        </div>
        <div
          class="c7"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tabs matches wrapper snapshot 1`] = `
<DocumentFragment>
  .c6 {
  position: absolute;
  top: 50%;
  right: 15px;
  width: 16px;
  height: 16px;
  margin-top: -8px;
}

.c6:after,
.c6:before {
  content: "";
  box-sizing: border-box;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: 25%;
  height: 25%;
  margin: auto;
  border: 2px solid transparent;
  background-color: #0b0c0c;
}

.c6:before {
  width: 100%;
}

.c6:after {
  height: 100%;
}

.c2 {
  border-top: 1px solid #bfc1c3;
  padding-top: 0;
}

.c3 {
  position: relative;
  padding-right: 40px;
  padding-bottom: 15px;
  cursor: pointer;
}

.c3:hover {
  background-color: #f8f8f8;
}

.c4 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 0;
  padding-top: 15px;
  padding-bottom: 0;
  padding-left: 0;
  border-width: 0;
  color: #005ea5;
  background: none;
  text-align: left;
  cursor: pointer;
}

.c5:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c5:focus {
  outline: none;
  background: none;
}

.c5:after {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  display: inline;
  border-width: 0;
  color: #005ea5;
  background: none;
  cursor: pointer;
}

.c1:link {
  color: #005ea5;
}

.c1:visited {
  color: #4c2c92;
}

.c1:hover {
  color: #2b8cc4;
}

.c1:active {
  color: #2b8cc4;
}

.c1:focus {
  color: #0b0c0c;
}

.c1:focus {
  outline: 3px solid #ffbf47;
  outline-offset: 0;
  background-color: #ffbf47;
}

.c1:focus {
  background: none;
}

.c0 {
  margin-bottom: 20px;
  border-bottom: 1px solid #bfc1c3;
}

@media (hover:none) {
  .c3:hover {
    background-color: initial;
  }
}

@media print {
  .c5 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media print {
  .c5 {
    font-family: sans-serif;
  }
}

@media print {
  .c1 {
    font-family: sans-serif;
  }
}

@media print {
  .c1[href^="/"]::after,
  .c1[href^="http://"]::after,
  .c1[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

<div
    class="c0"
  >
    <div
      style="text-align: right;"
    >
      <a
        class="c1"
        href="#"
      >
        Open all
      </a>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Writing well for the web
          </button>
        </h2>
        <div
          class="c6"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Writing well for specialists
          </button>
        </h2>
        <div
          class="c6"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            Know your audience
          </button>
        </h2>
        <div
          class="c6"
        />
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h2
          class="c4"
        >
          <button
            class="c5"
            type="button"
          >
            How people read
          </button>
        </h2>
        <div
          class="c6"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
